### PR TITLE
Fix empty offer location on old Vexl versions

### PR DESCRIPTION
**Pull request by SamTremko**

### CODE REVIEW

**Comment by SamTremko** [comment]
* **#929** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @SamTremko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/mobile/src/state/currentBtcPriceAtoms.ts [hunk 1]
```diff
@@ -75,9 +75,8 @@ export const refreshBtcPriceActionAtom = atom(
 
     const fetchInfo = get(btcPriceDataAtom)[currency]
     if (
-      fetchInfo?.state === 'loading' ||
```
Comment (SamTremko) [inline apps/mobile/src/state/currentBtcPriceAtoms.ts:78]: @kaladivo I think `loading` check is not needed here as `fetchInfo.lastRefresh` should do the same thing

File: apps/mobile/src/state/marketplace/atoms/offersToSeeInMarketplace.ts [hunk 2]
```diff
@@ -53,21 +53,25 @@ export const offersToSeeInMarketplaceAtom = atom((get) => {
...
-      // Not reported offers
-      !oneOffer.flags.reported &&
-      // Offers that has at least one common contact or are first degree
-      (oneOffer.offerInfo.privatePart.commonFriends.some((one) =>
-        importedContactsHashes.includes(one)
-      ) ||
-        oneOffer.offerInfo.privatePart.friendLevel.includes('FIRST_DEGREE')) &&
-      // Filter offers that are set to be in person but have no location
-      (isDeveloper ||
-        oneOffer.offerInfo.publicPart.locationState.includes('IN_PERSON') ||
```
Comment (SamTremko) [inline apps/mobile/src/state/marketplace/atoms/offersToSeeInMarketplace.ts:70]: I am not sure about this one, so correct me if I am wrong, but the old condition filtered out all `ONLINE` offers or? In new version I need also the `locationState.length = 0` to pass as there can be ONLINE offer without `locationState` set

Comment (kaladivo) [reply apps/mobile/src/state/marketplace/atoms/offersToSeeInMarketplace.ts:70]: The condition before location state was an array was: It should return true if the offer is online or if the length of specified locations is greater than 0

**Review by kaladivo — COMMENTED** [review]
I would prefer if we could go through this together

File: apps/mobile/src/components/OfferInfoPreview/components/ProductAndOtherOfferColumns.tsx [hunk 3]
```diff
@@ -21,40 +21,43 @@ function ProductAndOtherOfferColumns({offer}: Props): JSX.Element {
...
+                    <InfoText>{t('offerForm.pickup')}</InfoText>
+                  )}
+                  {offer.publicPart.locationState.includes('ONLINE') && (
+                    <InfoText>{t('offerForm.delivery')}</InfoText>
+                  )}
+                </YStack>
+              </InfoItemContainer>
+              <InfoDivider />
+            </>
+          ))}
```
Comment (kaladivo) [inline apps/mobile/src/components/OfferInfoPreview/components/ProductAndOtherOfferColumns.tsx]: Weird ')'

File: apps/mobile/src/state/marketplace/atoms/offersToSeeInMarketplace.ts [hunk 4]
```diff
@@ -53,21 +53,25 @@ export const offersToSeeInMarketplaceAtom = atom((get) => {
...
+        // Offers that has at least one common contact or are first degree
+        (oneOffer.offerInfo.privatePart.commonFriends.some((one) =>
+          importedContactsHashes.includes(one)
+        ) ||
+          oneOffer.offerInfo.privatePart.friendLevel.includes(
+            'FIRST_DEGREE'
+          )) &&
+        // Filter offers that are set to be in person but have no location
+        (isDeveloper ||
+          oneOffer.offerInfo.publicPart.locationState.length === 0)) ||
```
Comment (kaladivo) [inline apps/mobile/src/state/marketplace/atoms/offersToSeeInMarketplace.ts]: Check the last ')' if it should be there

**Review by kaladivo — APPROVED** [review]

**Comment by SamTremko** [comment]
### Merge activity

* **Mar 27, 2:01 PM EDT**: @SamTremko merged this pull request with [Graphite](https://app.graphite.dev/github/pr/vexl-it/vexl/929).

**Pull request merged**